pudb/debugger.py

```diff
@@ -878,17 +881,46 @@ def change_rhs_box(name, index, direction, w, size, key):
 
         # {{{ variables listeners
 
+        def get_inspect_info(id_path, read_only=False):
+            return (self.get_frame_var_info(read_only)
+                    .get_inspect_info(id_path, read_only))
+
+        def collapse_current(var, pos, iinfo):
+            if iinfo.show_detail:
+                # Simple case: collapse current variable
+                iinfo.show_detail = False
+            else:
+                # Complex case: collapse parent/container variable
+                desired_prefix = var.prefix[:-len(ValueWalker.PREFIX)]
+                try:
+                    # Walk backwards to find first shorter prefix
+                    parent = next(
+                        variable_widget
+                        for variable_widget in reversed(self.locals[:pos+1])
+                        if variable_widget.prefix == desired_prefix
+                    )
+                except StopIteration:
+                    # No parent found, so don't do anything
+                    pass
```

Comment:
Oh, one more thing: What do you think of popping up an error message here instead of silently swallowing the key? I'm not sure it's needed, but I figured I'd raise the question.

Reply:
Hmm... This case will be hit at the top level only. We don't give the user an error message when they do other similar ineffective operations, (e.g. pressing 'up' when they're already at the top of the variable list, or 'down' from the bottom of the breakpoint list), so I think I'd prefer to not popup an error.

That said, if 'h' is supposed to have some default behaviour across the right hand column, would returning the "key" variable at that point allow the default behaviour to trigger?

Reply:
That's fair. I'm OK with swallowing the key. Having it do something different (not that the key is currently mapped) depending on context would be confusing IMO. Thanks for considering it.